src/services/files/files.ts

```diff
@@ -28,6 +28,7 @@ export default (client: HttpClient, httpAuth: HttpInstance): FilesService => ({
 
     return (
       await client.post(httpAuth, '/', formData, {
+        onUploadProgress: options?.onUploadProgress,
```

Comment:
Just to be sure, we're not going to wrap the user provided callback in our own function to only forward the properties we want to share (`total` and `loaded`)?

Reply:
This is not intended.... I have destructured the event and declared the parameters in the callback. Awesome catch.